analysis/03_new_model_training/06_grid_To_mun_with_new_weightFile.md

```diff
@@ -0,0 +1,453 @@
+# Converting from grid-based to municipality-based
+
+The following steps were done to convert created grid_based dataset to municipality_based one:
+
+import the weight file to a dataframe
+
+assign the values to each grid 
+
+multiply the damaged values with the weights
+
+Aggregate the values by municipality and typhoon_name 
+
+```python
+%load_ext jupyter_black
+```
+
+```python
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+import matplotlib.pyplot as plt
+
+from sklearn import preprocessing
+from sklearn.preprocessing import RobustScaler
+from sklearn.model_selection import train_test_split
+from xgboost.sklearn import XGBRegressor
+from sklearn.metrics import mean_squared_error
+from math import sqrt
+
+from utils import get_training_dataset, weight_file
+```
+
+```python
+# Import the created dataset to a df
+df = get_training_dataset()
+df
+```
+
+```python
+# Read the weight CSV file and import to a df
+df_weight_m2c = weight_file("/phl_bld_weight_matrix.csv")
+df_weight_m2c.drop("Unnamed: 0", axis=1, inplace=True)
+
+# Convert id columns from float to int
+df_weight_m2c["grid_point_id"] = df_weight_m2c["id"].astype(np.int64)
+df_weight_m2c.drop(columns="id", inplace=True)
+df_weight_m2c
+```
+
+```python
+# check if weight_sum per municiplaity == 1
+agg_m2c = df_weight_m2c.groupby(["ADM3_PCODE"]).agg("sum")
+agg_m2c.sort_values("weight", ascending=True)
+```
+
+```python
+# values not summing up to 1
+agg_m2c[(agg_m2c["weight"] < 0.99)].sort_values("weight", ascending=True)
+```
+
+```python
+# Read the new weight CSV file and import to a df
+df_weight = weight_file("/phl_bld_grid_to_municip_matrix.csv")
+df_weight.drop("Unnamed: 0", axis=1, inplace=True)
+
+# Convert id columns from float to int
+df_weight["grid_point_id"] = df_weight["id"].astype(np.int64)
+df_weight.drop(columns="id", inplace=True)
+df_weight
+```
+
+```python
+# check if weight_sum per cell == 1
+agg_c2m = df_weight.groupby(["grid_point_id"]).agg("sum")
+agg_c2m.sort_values("weight", ascending=True)
+```
+
+```python
+# values not summing up to 1
```

Comment:
@PaulineNimo it seems that there are ~200 grid points with weights that don't sum to 1, could you check why this is happening?

Reply:
Thanks so much Pauline! Let @mkooshki know once this dataset is updated.